storybookjs/storybook
# UI: Add ability to programmatically hide the toolbar

Closes #22081 


## What I did

Following conversations on the issue, I:
* Added a manager option to add a function that controls the toolbar visibility
* Edited the preview code to handle it
* Added stories to test/document toolbar visibility behaviour
* Edited documentation to mention the new function

I have some doubts over the following:
* I had to seriously hack the story showing the toolbar using my new option, because `addons` is global and the FakeProvider can't be made to change it
* I am not sure about documentation standards (tone of voice, glossary, etc)
* I chose not to document all of `layout` in my function parameters as I believe some props to be irrelevant, or possibly legacy (`showTabs`)
* I didn't merge my new documentation into the `Toolbars and Globals` page because it's already cramped, and answers to creating custom menus; it felt more logical to have a page dedicated to the Toolbar UI element with the same documentation style as the sidebar, etc



## How to test

1. Customise `code/ui/.storybook/manager.tsx` with the code snippet below
2. Run `yarn storybook:ui`
3. Open Storybook in your browser
4. Notice how the toolbar is gone

```ts
addons.setConfig({
  toolbar: { showToolbar: () => false },
});
```


## Checklist

      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)~

#### Maintainers


`["cleanup", "BREAKING CHANGE", "feature request", "bug", "build", "documentation", "maintenance", "dependencies", "other"]`
